src/handlers/events/delivery/build/BuildOnScanSuccessStatus.ts

```diff
@@ -51,22 +61,41 @@ export class BuildOnScanSuccessStatus implements StatusSuccessHandler {
             siblings: status.commit.statuses,
         };
 
-        if (nothingFailed(statusAndFriends) && !previousPhaseSucceeded(params.phases, this.ourContext, statusAndFriends)) {
+        logger.debug(`BuildOnScanSuccessStatus: our context=[%s], %d conditional builders, statusAndFriends=[%j]`,
+            params.ourContext, params.conditionalBuilders.length, statusAndFriends);
+
+        if (nothingFailed(statusAndFriends) && !previousPhaseSucceeded(params.phases, params.ourContext, statusAndFriends)) {
             return Success;
         }
 
-        if (!currentPhaseIsStillPending(this.ourContext, statusAndFriends)) {
+        if (!currentPhaseIsStillPending(params.ourContext, statusAndFriends)) {
             return Success;
         }
 
         logger.info(`Running build. Triggered by ${status.state} status: ${status.context}: ${status.description}`);
-        await dedup(commit.sha, () => {
+        await dedup(commit.sha, async () =>  {
             const id = new GitHubRepoRef(commit.repo.owner, commit.repo.name, commit.sha);
-            const creds = {token: params.githubToken};
+            const credentials = {token: params.githubToken};
+            const project = await GitCommandGitProject.cloned(credentials, id);
+
+            const i: ProjectListenerInvocation = {
+                id,
+                project,
+                credentials,
+                context,
+                addressChannels: addressChannelsFor(commit.repo, context),
+            };
+
+            const builders: boolean[] = await Promise.all(params.conditionalBuilders.map(b => b.test(i)));
+            const indx = builders.indexOf(true);
```

Comment:
what about ...
`const builder = conditionalBuilders.find(async (a) => a.test(i)) `

Reply:
Thanks, I'll change it after merge